astropy/cosmology/parameter.py

```diff
@@ -332,17 +349,20 @@ def _validate_to_float(cosmology, param, value):
 
 @Parameter.register_validator("scalar")
 def _validate_to_scalar(cosmology, param, value):
-    """"""
+    """Parameter value validator where value is a scalar."""
     value = _validate_with_unit(cosmology, param, value)
-    if not value.isscalar:
-        raise ValueError(f"{param.name} is a non-scalar quantity")
-    return value
+
+    # Check it's a scalar (Quantity or other)
+    if (hasattr(value, "isscalar") and value.isscalar) or np.isscalar(value):
```

Comment:
I thought `np.isscalar` might do the first part, but this turns out not to be the case. The docstring, though, suggests `np.ndim(value) == 0` as an alternative that would do what you want and is shorter.

Reply:
Thanks!